src/gui/transferlistsortmodel.cpp

```diff
@@ -219,16 +236,11 @@ bool TransferListSortModel::lessThan(const QModelIndex &left, const QModelIndex
 {
     Q_ASSERT(left.column() == right.column());
 
-    if (m_lastSortColumn != left.column())
-    {
-        if (m_lastSortColumn != -1)
-            m_subSortColumn = m_lastSortColumn;
-        m_lastSortColumn = left.column();
-    }
-
     const int result = compare(left, right);
-    if (result == 0)
-        return compare(left.sibling(left.row(), m_subSortColumn), right.sibling(right.row(), m_subSortColumn)) < 0;
+    if (result == 0) {
+        const int subResult = compare(left.sibling(left.row(), m_subSortColumn), right.sibling(right.row(), m_subSortColumn));
+        return m_subSortColumnOrder == m_lastSortColumnOrder ? subResult < 0 : subResult > 0;
```

Comment:
Why do you compare `m_subSortColumnOrder` with `m_lastSortColumnOrder`? It's not clear. At least you should add a code comment.

Reply:
Hm... Actually this is wrong. Looks like I've outsmarted myself. Fixed, comment added

Reply:
@glassez 
Well, I was wrong last time but right first time. Qt reverses `lessThan()` as a whole so sorting descending means that sub-
sorting has to be reversed. And if sub-sorting is descending the sub-result must be inversed too. That means if both ordered descending the sub-result is double-inversed (which is the same as no inversion). Obviously, if both ordered Ascending this also means that no inversion should happen. And to check that exactly one is descending is the same as checking that both are not equal.
Fixed, comment added.